### ISSUE TITLE
Does the webhook URL have to be under the domain "*.ngrok.io"?

### ISSUE THREAD

**Issue by MingZhang-PS** [RecastAI/starter-NodeJS#7]
I tried to deploy my webhook on SCP cloud platform, the webhook URL looked like https://ming-chatapp-demo.cfapps.sap.hana.ondemand.com/respone. But it is strange that there is no request come into my webhook

**Comment by dbousque** [comment]
Hi @MingZhang860612 :) Webhooks don't have to to be under the domain "ngrok.io", the only constraint is that it's `https` with a valid certificate. The request made is a POST, maybe your server only handles GET for this route ? If you still have issues, could you please invite me as a collaborator to your bot (my nickname is 'dominik') and show me an exemple of a conversation where the problem happens ?

**Comment by TechSkieur** [comment]
Hi @dbousque , I added you as a collaborator into my bot, you may go to the skill greetings and find the Actions I defined. I try to use a webhook to call API: https://recast-integration-proxy.cfapps.sap.hana.ondemand.com/dialog, this API only supports POST method. But based on our CF logs, the request does not come to the service. In postman, this API is working correctly. 

Thanks,

yangyang

**Comment by dbousque** [comment]
Hi @TechSkieur and @MingZhang860612. It seems to be linked to an SAP network issue. Try this request inside and outside of SAP's network : 
```
curl -X POST -H 'Content-Type: application/json' https://recast-integration-proxy.cfapps.sap.hana.ondemand.com/dialog -d '{}'
```
For me, it's working inside (HTTP 400) but not outside (crash). This is pretty much the request that the bot-builder makes to your server. Do you know what might happen ?

**Comment by TechSkieur** [comment]
@dbousque Yes, I noticed this issue as well. I'm not sure whether it's due to the Canary landscape of SCP. I throw a ticket to them. Thanks.

yangyang

**Comment by anil541992** [comment]
Hi,
@MingZhang860612 Did you find a solution to the above issue?.Even i am facing the same issue,kindly help
regards,
Anil

**Comment by dbousque** [comment]
Try to query your server from inside and outside of SAP's network, you will probably see that your server is not available on the internet.

**Comment by anil541992** [comment]
Thanks a lot for your response!!
Yes you are absolutely right,how do i proceed ?

**Comment by dbousque** [comment]
I don't know, I suppose there should be a way to make a server hosted on SCP cloud platform available on the internet, but maybe they want to prevent that for now.
